tests/test_modeling_common.py

```diff
@@ -60,6 +72,33 @@ def test_cached_files_are_used_when_no_internet(self):
             if p1.data.ne(p2.data).sum() > 0:
                 assert False, "Parameters not the same!"
 
+    def test_one_request_upon_cached(self):
+        import diffusers
+
+        diffusers.utils.import_utils._safetensors_available = False
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            with requests_mock.mock(real_http=True) as m:
+                UNet2DConditionModel.from_pretrained(
+                    "hf-internal-testing/tiny-stable-diffusion-torch", subfolder="unet", cache_dir=tmpdirname
+                )
+
+            download_requests = [r.method for r in m.request_history]
+            assert download_requests.count("HEAD") == 2, "2 HEAD requests one for config, one for model"
+            assert download_requests.count("GET") == 2, "2 GET requests one for config, one for model"
+
+            with requests_mock.mock(real_http=True) as m:
+                UNet2DConditionModel.from_pretrained(
+                    "hf-internal-testing/tiny-stable-diffusion-torch", subfolder="unet", cache_dir=tmpdirname
+                )
+
+            cache_requests = [r.method for r in m.request_history]
+            assert (
+                "HEAD" == cache_requests[0] and len(cache_requests) == 1
+            ), "We should call only `model_info` to check for _commit hash and `send_telemetry`"
+
+        diffusers.utils.import_utils._safetensors_available = True
+
```

Comment:
```suggestion
    def test_one_request_upon_cached(self):
        try:
            diffusers.utils.import_utils._safetensors_available = False

            with tempfile.TemporaryDirectory() as tmpdirname:
                with requests_mock.mock(real_http=True) as m:
                    UNet2DConditionModel.from_pretrained(
                        "hf-internal-testing/tiny-stable-diffusion-torch", subfolder="unet", cache_dir=tmpdirname
                    )

                download_requests = [r.method for r in m.request_history]
                assert download_requests.count("HEAD") == 2, "2 HEAD requests one for config, one for model"
                assert download_requests.count("GET") == 2, "2 GET requests one for config, one for model"

                with requests_mock.mock(real_http=True) as m:
                    UNet2DConditionModel.from_pretrained(
                        "hf-internal-testing/tiny-stable-diffusion-torch", subfolder="unet", cache_dir=tmpdirname
                    )

                cache_requests = [r.method for r in m.request_history]
                assert (
                    "HEAD" == cache_requests[0] and len(cache_requests) == 1
                ), "We should call only `model_info` to check for _commit hash and `send_telemetry`"

        finally: 
            diffusers.utils.import_utils._safetensors_available = True
```

I think you're trying to ensure that the _safetensors available gets reset at the end of the function. Ideally we use a mocking library but if that's outside the scope of the PR, we can just put a finally block at the end of the function to ensure it gets reset.

Reply:
Don't want to put try-finally in tests, do not think this is good practice. We can remove this safetensors is available logic once: https://github.com/huggingface/diffusers/pull/2123 is merged.